app/services/forest_liana/forest_api_requester.rb

```diff
@@ -1,33 +1,53 @@
+require 'httparty'
+
 module ForestLiana
   class ForestApiRequester
-    def perform_request(query_parameters = nil)
-      http = Net::HTTP.new(@uri.host, @uri.port)
-      http.use_ssl = true if forest_api_url.start_with?('https')
-
-      http.start do |client|
-        path = get_path(query_parameters)
-        request = Net::HTTP::Get.new(path)
-        request['Content-Type'] = 'application/json'
-        request['forest-secret-key'] = ForestLiana.env_secret
-        request['forest-token'] = @forest_token if @forest_token
-        response = client.request(request)
-
-        handle_service_response(response)
+    def self.get(route, query: nil, headers: {})
+      begin
+        response = HTTParty.get("#{forest_api_url}#{route}", {
+          headers: base_headers.merge(headers),
+          query: query,
+        }).response
+      rescue
+        raise 'Cannot reach Forest API, it seems to be down right now.'
       end
+
+      check_configuration_errors(response)
+    end
+
+    def self.post(route, body: nil, query: nil, headers: {})
+      response = HTTParty.post("#{forest_api_url}#{route}", {
+        headers: base_headers.merge(headers),
+        query: query,
+        body: body.to_json,
+      }).response
+
+      check_configuration_errors(response)
     end
 
-    def forest_api_url
-      ENV['FOREST_URL'] || 'https://api.forestadmin.com';
+    private
+
+    def self.base_headers
+      {
+        'Content-Type' => 'application/json',
+        'forest-secret-key' => ForestLiana.env_secret,
+      }
+    end
+
+    def self.forest_api_url
+      ENV['FOREST_URL'] || 'https://api.forestadmin.com'
     end
 
-    def get_path(query_parameters)
-      route = @uri.path
-      unless query_parameters.nil?
-        query = query_parameters
-          .collect { |parameter, value| "#{parameter}=#{CGI::escape(value.to_s)}" }.join('&')
-        route += "?#{query}"
+    def self.check_configuration_errors(response)
+      if response.is_a?(Net::HTTPNotFound)
+        error_message = 'Cannot retrieve the data from the Forest server. Can you check that you properly copied the Forest envSecret in the Liana initializer?'
+      elsif response.is_a?(Net::HTTPUnprocessableEntity)
+        error_message = 'Cannot retrieve the project you\'re trying to unlock. The envSecret and renderingId seems to be missing or inconsistent.'
       end
-      route
+
+      raise Errors::HTTP401Error.new(error_message) if error_message
```

Comment:
For a generic service, I am not sure it is a good idea to respond with a 401. Depending on the usage not sure we want to disconnect the end user.